src/API/Logs/LogsClient.php

```diff
@@ -58,4 +62,59 @@ public function getBuildLog(string $accessToken, string $sappId)
         return $this->guzzle($this->getBearerTokenMiddleware($accessToken))
             ->get("sapps/{$sappId}/build-logs");
     }
+
+    /**
+     * @var $timeout in seconds, 0 means no timeout
+     */
+    public function streamLog(string $accessToken, callable $sender, array $topic, int $timeout = 0, OutputInterface $output = null)
+    {
+        $this->topic = $topic;
+        $this->sender = $sender;
+        $this->accessToken = $accessToken;
+        $this->timeout = $timeout;
+        $output = function($msg) use($output) {
+            if ($output !== null) {
+                $output->writeln($msg);
+            }
+        };
+
+        $uri = str_replace('http', 'ws', $this->baseUrl);
+        $uri = rtrim($uri, '/');
+        $uri .= '/logs';
+
+        $loop = \React\EventLoop\Factory::create();
+        $reactConnector = new \React\Socket\Connector($loop);
+        $connector = new \Ratchet\Client\Connector($loop, $reactConnector);
+
+        $connector($uri)
+            ->then(function(Ratchet\Client\WebSocket $conn) use ($loop, $output) {
+
+                if ($this->timeout > 0) {
+                    $loop->addTimer(2, function() use($conn) {
```

Comment:
Is this supposed to be the value of `$this->timeout` instead of `2`?

Reply:
it is, nice catch